src/Agent.Plugins/TfsVCSourceProvider.cs

```diff
@@ -253,7 +253,8 @@ public sealed class TfsVCSourceProvider : ISourceProvider
                 tfWorkspaces = await tf.WorkspacesAsync(matchWorkspaceNameOnAnyComputer: true);
                 foreach (ITfsVCWorkspace tfWorkspace in tfWorkspaces ?? new ITfsVCWorkspace[0])
                 {
-                    await tf.WorkspaceDeleteAsync(tfWorkspace);
```

Comment:
Please add message trace for the case when tf.WorkspaceDeleteAsync returns false.

Reply:
Nvm, the message is already here in WorkspaceDeleteAsync